src/Scope/Subst.hs

```diff
@@ -126,11 +165,13 @@ subst s a = runSubst s (substM a)
 
 -- Run substM under a given context
 substWithCtxt :: Substitutable a => Ctxt -> Subst -> a -> a
-substWithCtxt g s = subst (Map.union (Map.mapWithKey (const . SubVar) g) s)
+substWithCtxt g = subst . remember (C.tmVars  g)
```

Comment:
This means that local variables won't be alpha-renamed apart from either global defines or datatype names. But I guess I don't see why that would be a problem.